crates/s3s/src/ops/signature.rs

```diff
@@ -276,24 +276,27 @@ impl SignatureContext<'_> {
                     sig_v4::Payload::Empty
                 };
                 sig_v4::create_canonical_request(method, uri_path, query_strings, &headers, payload)
+            } else if matches!(amz_content_sha256, AmzContentSha256::UnsignedPayload) {
```

Comment:
I forget the reason since this code block is at least three years old.
You can try to simplify the signature logic and see if it passes the e2e test. I'll also check it.